File: client/src/dto/graphDto/GraphDataDto.ts
import {NodeDataDto} from "./NodeDataDto";
import {EdgeDataDto} from "./EdgeDataDto";


/**
 * Сущность данных, пришедшая от сервера, должна быть сконвертирована в сущность клиента
 */
export interface GraphDataDto {
    nodes: NodeDataDto[],
    edges: EdgeDataDto[]
}

File: server/src/main/kotlin/knowledgegraph/server/dto/graph/GraphDataDto.kt
package knowledgegraph.server.dto.graph

import knowledgegraph.server.entity.GraphData

data class GraphDataDto(
    val nodes: List<NodeDataDto>,
    val edges: List<EdgeDataDto>
) {
    fun toGraphData() = GraphData(
        nodes = nodes.map { it.toNodeData() },
        edges = edges.map { it.toEdgeData() }
    )
}